app/ide-desktop/lib/content/src/index.ts

```diff
@@ -148,75 +160,84 @@ class Main {
+            contentConfig.OPTIONS.groups.startup.options.entry.default
+        if ((isUsingAuthentication || isUsingNewDashboard) && isOpeningIde) {
+            const hideAuth = () => {
+                const auth = document.getElementById('dashboard')
+                const ide = document.getElementById('root')
+                if (auth) {
+                    auth.style.display = 'none'
+                }
+                if (ide) {
+                    ide.hidden = false
+                }
+            }
+            /** This package is an Electron desktop app (i.e., not in the Cloud), so
+             * we're running on the desktop. */
+            /** TODO [NP]: https://github.com/enso-org/cloud-v2/issues/345
+             * `content` and `dashboard` packages **MUST BE MERGED INTO ONE**. The IDE
+             * should only have one entry point. Right now, we have two. One for the cloud
+             * and one for the desktop. Once these are merged, we can't hardcode the
+             * platform here, and need to detect it from the environment. */
+            const currentPlatform = contentConfig.OPTIONS.groups.startup.options.platform.value
+            const webPlatform = contentConfig.OPTIONS.groups.startup.options.platform.default
+            let platform = authentication.Platform.desktop
+            if (currentPlatform === webPlatform) {
+                platform = authentication.Platform.cloud
             }
+            /** FIXME [PB]: https://github.com/enso-org/cloud-v2/issues/366
+             * React hooks rerender themselves multiple times. It is resulting in multiple
+             * Enso main scene being initialized. As a temporary workaround we check whether
+             * appInstance was already ran. Target solution should move running appInstance
+             * where it will be called only once. */
+            let appInstanceRan = false
+            const onAuthenticated = () => {
+                if (!contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value) {
+                    hideAuth()
+                    if (!appInstanceRan) {
+                        appInstanceRan = true
+                        void this.runApp()
+                    }
+                }
+            }
+            authentication.run({
+                appRunner: this,
+                logger,
+                platform,
+                showDashboard:
+                    contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value,
+                onAuthenticated,
+            })
         } else {
-            console.error('Failed to initialize the application.')
+            void this.runApp()
         }
     }
 }
 
-const API = new Main()
-
-// @ts-expect-error `globalConfig.windowAppScopeName` is not known at typecheck time.
-// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-window[GLOBAL_CONFIG.windowAppScopeName] = API
```

Comment:
What happened with this line? We are accessing the JS app from Rust via this variable. Why was it deleted? Was it moved somewhere? Such changes as this should be very clearly explained as they can break a lot of things.

Reply:
@wdanilo are you sure that rust uses this global? this assignes Main to `window.enso` which I can't find any references in code to be used.

App also expose `ensoglApp` in window https://github.com/enso-org/enso/blob/develop/lib/rust/ensogl/pack/js/src/runner/index.ts#L217 which is widely accessed e.g. in `gui/analytics/src/remote_log.js`. It also has a comment that rust access it.

Reply:
oh yeah - the initial commit (the revert commit) already had the global removed so all QAs seemed to work fine without the global.
i added it back because i figured it's better to take the ide team's word for it

Reply:
Please, never remove lines that you are not 100% sure can be removed. If you want to refactor such a code, the PR description should have a very explicit information about it, so reviewers can be extra careful when checking this place. This removed line was used in Rust in several places, so this change was introducing regression.